cpp/test/sg/decisiontree_batchedlevel_unittest.cu

```diff
@@ -0,0 +1,227 @@
+    // Nodes that are created new by the invocation of nodeSplitKernel()
+    new_nodes = static_cast<NodeT*>(
+      d_allocator->allocate(sizeof(NodeT) * 2 * max_batch, 0));
+    // Number of nodes and leaves that are created new by the invocation of
+    // nodeSplitKernel()
+    n_new_nodes = static_cast<IdxT*>(d_allocator->allocate(sizeof(IdxT), 0));
+    n_new_leaves = static_cast<IdxT*>(d_allocator->allocate(sizeof(IdxT), 0));
+    // New depth reached by the invocation of nodeSplitKernel()
+    new_depth = static_cast<IdxT*>(d_allocator->allocate(sizeof(IdxT), 0));
+
+    splits = static_cast<Traits::SplitT*>(
+      d_allocator->allocate(sizeof(Traits::SplitT) * max_batch, 0));
+
+    raft::update_device(data, h_data.data(), n_row * n_col, 0);
+    raft::update_device(labels, h_labels.data(), n_row, 0);
+    MLCommon::iota(row_ids, 0, 1, n_row, 0);
+    MLCommon::iota(col_ids, 0, 1, n_col, 0);
+
+    tempmem = std::make_shared<TemporaryMemory<DataT, LabelT>>(
+      *raft_handle, cudaStream_t(0), n_row, n_col, 0, params);
+    preprocess_quantile(data, reinterpret_cast<unsigned int*>(row_ids), n_row,
+                        n_col, n_row, n_bins, tempmem);
+    DataT* quantiles = tempmem->d_quantile->data();
+    CUDA_CHECK(cudaStreamSynchronize(0));
+
+    h_quantiles = tempmem->h_quantile->data();
+
+    input.data = data;
+    input.labels = labels;
+    input.M = n_row;
+    input.N = n_col;
+    input.nSampledRows = n_row;
+    input.nSampledCols = n_col;
+    input.rowids = row_ids;
+    input.colids = col_ids;
+    input.nclasses = 0;  // not applicable for regression
+    input.quantiles = quantiles;
+  }
+
+  void TearDown() override {
+    auto d_allocator = raft_handle->get_device_allocator();
+    d_allocator->deallocate(data, sizeof(DataT) * n_row * n_col, 0);
+    d_allocator->deallocate(labels, sizeof(LabelT) * n_row, 0);
+    d_allocator->deallocate(row_ids, sizeof(IdxT) * n_row, 0);
+    d_allocator->deallocate(col_ids, sizeof(IdxT) * n_col, 0);
+    d_allocator->deallocate(curr_nodes, sizeof(NodeT) * max_batch, 0);
+    d_allocator->deallocate(new_nodes, sizeof(NodeT) * 2 * max_batch, 0);
+    d_allocator->deallocate(n_new_nodes, sizeof(IdxT), 0);
+    d_allocator->deallocate(n_new_leaves, sizeof(IdxT), 0);
+    d_allocator->deallocate(new_depth, sizeof(IdxT), 0);
+    d_allocator->deallocate(splits, sizeof(Traits::SplitT) * max_batch, 0);
+  }
+
+  DecisionTreeParams params;
+
+  std::unique_ptr<raft::handle_t> raft_handle;
+  std::shared_ptr<TemporaryMemory<DataT, LabelT>> tempmem;
+
+  DataT* h_quantiles;
+  Traits::InputT input;
+
+  NodeT* curr_nodes;
+  NodeT* new_nodes;
+  IdxT* n_new_nodes;
+  IdxT* n_new_leaves;
+  IdxT* new_depth;
+  Traits::SplitT* splits;
+
+  DataT* data;
+  DataT* labels;
+  IdxT* row_ids;
+  IdxT* col_ids;
+};
+
+TEST_F(NodeSplitKernelUnitTest, Quantiles) {
+  /* Ensure that quantiles are computed correctly */
+  std::vector<DataT> expected_quantiles[]{{-2.0f, -1.0f, 0.0f, 2.0f},
```

Comment:
Nice! Would be great to extend this to any funky corner cases that come to mind - like quantiles with ties (like 50% of the data has the same value)

Reply:
Sure, let me think about possible extensions. For this toy example, I wanted to make sure that all values are being picked up by quantile sketcher, since the example is very tiny.

Reply:
As per the other example, I'm on board moving that to a later PR.